### DISCUSSION TITLE
Rstudio / gt / fmt_currency

### DISCUSSION THREAD

**Discussion by cuentodeadas** [rstudio/gt#1919] [Q&A] [answered]
Hello, this is my first time posting - hopefully I'm in the right place.
**I am having a hard time figuring out how to simply add a dollar sign in front of my average wage figures in the below table of wages by gender. I only want the dollar sign in front of the wage figure, of course, not the #n in parentheses.** 
I also wish I could just add some text under the "Average Wage" that reads "(# paid hourly)" instead of in a footnote. 
Welcome any other feedback as well. Many thanks!

```
WS_subset_data %>%
  filter(!is.na(x60_Wage2_Fix)) %>%
  tbl_continuous(
    variable = x60_Wage2_Fix,
    statistic = ~"{mean} ({N_nonmiss})",
    by = x24_SurveyType2,
    include = c(x37_Gender_recode, x1Dummy_var),
    value = x1Dummy_var ~ 1,
    label = list(
      x37_Gender_recode = "By Gender",
      x1Dummy_var = "Overall")
  ) %>%
  modify_header(
    label = "",
    all_stat_cols() ~ "**Average Wage**"
  ) %>%
  modify_footnote(all_stat_cols() ~ "*(# paid hourly)*") %>%
  as_gt() %>%  
  fmt_currency(columns = everything(),
               currency = "dollar")
gt_theme_espn()
```

![image](https://github.com/user-attachments/assets/2bb72e56-61d4-4a69-a743-517c975c9032)

**Comment by rich-iannone** [comment, marked as answer]
While the supplied code isn't reproducible (`WS_subset_data` isn't available here), we can work with a similar gtsummary example.

This table can be made with `gtsummary` and transformed into a gt table:

```r
library(gtsummary)
library(gt)

gt_tbl <- 
  tbl_summary(
  trial,
  include = c(age, grade, response),
  by = trt, # split table by group
  missing = "no" # don't list missing data separately
) |> 
  add_n() |> # add column with total number of non-missing observations
  add_p() |> # test for a difference between groups
  modify_header(label = "**Variable**") |> # update the column header
  bold_labels() |>
  as_gt()

gt_tbl
```

<img width="711" alt="image" src="https://github.com/user-attachments/assets/e5fc96b2-496c-4ccf-a766-38b6ba58fe1e">

One problem with working with this table is that we have no idea what the underlying column names are. But we can use gt's `tab_info()` to find out:

```r
# Use `tab_info()` to get the column names
tab_info(gt_tbl)
```

<img width="714" alt="image" src="https://github.com/user-attachments/assets/52191681-2c2e-4f7e-a334-41bec8aef579">

It looks like the `stat_*` columns are what we want to change. However, we can't format them with `fmt_currency()` because  gtsummary creates a table of character-based columns (you need numeric-type columns to do any number-based formatting). But you can use `text_transform()` to apply a function to the text-based values.

```r
gt_tbl |>
  text_transform(
    fn = function(x) paste0("$", x),
    locations = cells_body(
      columns=starts_with("stat"),
      rows=c(1, 3, 4, 5, 6))
  )
```

<img width="714" alt="image" src="https://github.com/user-attachments/assets/2bf5645f-8147-4cd6-b7fd-dfbd56eb0cc6">

I hope this provides enough info for fixing your formatting issue!

**Comment by rich-iannone** [comment]
To answer the second part about the footer note. You can remove the existing footnote with gt's `rm_footnotes()` and apply your own with `tab_source_note()`. Here's a continuation of the above example:

```r
gt_tbl |>
  text_transform(
    fn = function(x) paste0("$", x),
    locations = cells_body(
      columns=starts_with("stat"),
      rows=c(1, 3, 4, 5, 6))
  ) |>
  rm_footnotes() |>
  tab_source_note("My custom footer note.")
```

<img width="711" alt="image" src="https://github.com/user-attachments/assets/d495cbb8-eb7a-4794-9347-9886669c9942">

**Reply by cuentodeadas** [reply]
Thank you these worked perfectly!
